app/app.go

```diff
@@ -63,6 +63,66 @@ func New(options ...AppOption) *App {
 	return app
 }
 
+func (a *App) InitServer() {
+	a.srv.AppInitialized.Do(func() {
+		a.initEnterprise()
+		a.accountMigration = a.srv.AccountMigration
+		a.ldap = a.srv.Ldap
+		a.notification = a.srv.Notification
+		a.saml = a.srv.Saml
+
+		a.StartPushNotificationsHubWorkers()
+		a.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
+			if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
+				if appErr := a.DeactivateGuests(); appErr != nil {
+					mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
+				}
+			}
+		})
+
+		// Disable active guest accounts on first run if guest accounts are disabled
+		if !*a.Config().GuestAccountsSettings.Enable {
+			if appErr := a.DeactivateGuests(); appErr != nil {
+				mlog.Error("Unable to deactivate guest accounts", mlog.Err(appErr))
+			}
+		}
+
+		if a.srv.joinCluster && a.srv.Cluster != nil {
+			a.registerAllClusterMessageHandlers()
+			a.srv.Cluster.StartInterNodeCommunication()
+		}
+
+		pluginsRoute := a.srv.Router.PathPrefix("/plugins/{plugin_id:[A-Za-z0-9\\_\\-\\.]+}").Subrouter()
+		pluginsRoute.HandleFunc("", a.ServePluginRequest)
+		pluginsRoute.HandleFunc("/public/{public_file:.*}", a.ServePluginPublicRequest)
+		pluginsRoute.HandleFunc("/{anything:.*}", a.ServePluginRequest)
+		a.srv.Router.NotFoundHandler = http.HandlerFunc(a.Handle404)
+
+		if model.BuildEnterpriseReady == "true" {
+			a.LoadLicense()
+		}
+
+		a.DoAppMigrations()
+
+		a.InitPostMetadata()
+
+		a.InitPlugins(*a.Config().PluginSettings.Directory, *a.Config().PluginSettings.ClientDirectory)
+		a.AddConfigListener(func(prevCfg, cfg *model.Config) {
+			if *cfg.PluginSettings.Enable {
+				a.InitPlugins(*cfg.PluginSettings.Directory, *a.Config().PluginSettings.ClientDirectory)
+			} else {
+				a.srv.ShutDownPlugins()
+			}
+		})
+		a.initJobs()
+		a.srv.RunJobs()
+	})
+	a.accountMigration = a.srv.AccountMigration
+	a.ldap = a.srv.Ldap
+	a.notification = a.srv.Notification
+	a.saml = a.srv.Saml
+}
+
 // DO NOT CALL THIS.
```

Comment:
Unrelated, but can we also file a ticket to fix this sometime ? Would be good to get it done.

Reply:
sure we should

Reply:
Filed: https://mattermost.atlassian.net/browse/MM-25744